### PR TITLE
ARM64 support.

### PR DESCRIPTION
**Pull request by sas**

### CODE REVIEW

**Comment by strager** [comment]
lgtm

File: Definitions/arm64.json [hunk 1]
```diff
@@ -0,0 +1,776 @@
...
+                "format"                : "hex",
+
+                "base-gdb-reg-number"   : 0,
+                "base-gcc-reg-number"   : 0,
+                "base-dwarf-reg-number" : 0,
+
+                "referencing-sets"      : [ "pseudo-gpr32" ]
+            },
+
+            "r0"  : {
```
Comment (chancila) [inline Definitions/arm64.json]: why r instead of x with r aliases?

**Comment by strager** [comment]
It seems an AArch64 ds2 will not be able to debug an ARM program. Is that correct and intended?

**Comment by strager** [comment]
lgtm, but the gdb special case is confusing to me.

**Comment by strager** [comment]
(Sorry for the delay in reviewing.)

**Comment by sas** [comment]
Yes.

On Wed, Apr 1, 2015 at 5:49 PM, strager <notifications@github.com> wrote:

> It seems an AArch64 ds2 will not be able to debug an ARM program. Is that
> correct and intended?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/ds2/pull/19#issuecomment-88676283>.
>



-- 
Stephane Sezer

**Comment by sas** [comment]
As far as I can see there's been only two comments on this branch: x vs r for register names, which sounds pretty arbitrary, and the gdb special case thing. I'll take a look at the gdb code path a bit later. For now it's just mimicking whatever we have for arm already and I'd rather checkin this branch before it goes stale.

**Pull request closed without merging**